core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt

```diff
@@ -130,15 +133,14 @@ class Falstad(val source: String) {
 		if(DEBUG) {
 			val repmap: MutableMap<PosSet, MutableSet<PosSet>> = mutableMapOf()
 			roots.values.forEach {
-				dprintln("F.<init>: r $it => ${it.representative}")
+				dprintln("F.<init>: r ${it} => ${it.representative}")
 				repmap.getOrPut(it.representative as PosSet, { mutableSetOf() }).add(it)
 			}
 
 			repmap.entries.forEach {
 				dprintln("F.<init>: R ${it.key}:")
 				it.value.forEach {
-					it2 ->
-					dprintln(" - $it2")
+					dprintln(" - $it")
```

Comment:
revert

Reply:
This change is because the `it` variable shadowed the larger scope's variable, and I've seen that explode in fantastically nondeterministic ways at least in my experience.

Reply:
It very much shouldn't explode; the lexical scoping here should be sensible. One does have to take care to note that the outer `it` is indeed inaccessible, or change the name via a named parameter.

If the change were a good idea, I'd certainly prefer a more descriptive name than `it2`.

Reply:
How about we just avoid the problem entirely and name it properly? I couldn't come up with a good name at the time.